lib/Dialect/TritonGPU/Transforms/ReorderInstructions.cpp

```diff
@@ -97,6 +99,8 @@ class TritonGPUReorderInstructionsPass
       auto BOp = dotUser.getOperand(1).getDefiningOp();
       if (!BOp)
         return;
+      if (!dom.dominates(op.getOperand(), BOp))
```

Comment:
```
%b = cvt %cst1
%a = cvt %cst0
dot %a, %b
```
In this case, is `dom.dominstates(%cst0, %b) == False`? I think we could still reorder `%a` and `%b` right?

Reply:
In this case it would still re-order as %cst0 dominates %b (unless %cst0 is in between the ops).
One thing that could be done is to move the idx 1 cvt down instead of moving the idx 0 cvt but I'm not sure if that's what we want?

Reply:
Oh, I got it. Probably I understood it wrong, in SSACFG,`%a` dominates `%b` as long as `%a` is defined before `%b`, if they are in the same block.

Yeah, so there are two cases
```
# case1: %cst0 = ...
%b = cvt %cst1
# case2: %cst0 = ...
%a = cvt %cst0
dot %a, %b
```
The current implementation fails for case2 if I understand it correct. If we move `%b` after `%a`, that seems better.

Reply:
right, it would be easy to change the logic. Do you think this is a better solution? I'm not sure why it is done that way in a first place

Reply:
I do think so, but we can confirm it with Phil when he is back